lib/utils/txrejectederror.js

```diff
@@ -0,0 +1,15 @@
+var inherits = require("util").inherits;
+
+// raised when the transaction is rejected prior to running it in the EVM.
+function TXRejectedError(message) {
```

Comment:
Creating this type wasn't absolutely necessary in the end, but I thought it helped to express the intent behind the errors being returned, so I left it in.